Neos.Media/Classes/Domain/Model/ThumbnailGenerator/DocumentThumbnailGenerator.php

```diff
@@ -64,22 +64,26 @@ public function refresh(Thumbnail $thumbnail)
             $im->setImageFormat('png');
             $im->setImageBackgroundColor('white');
             $im->setImageCompose(\Imagick::COMPOSITE_OVER);
+            
+            if (method_exists($im, 'mergeImageLayers')) {
+                // Replace flattenImages in imagick 3.3.0
+                // @see https://pecl.php.net/package/imagick/3.3.0RC2
+                $im = $im->mergeImageLayers(\Imagick::LAYERMETHOD_MERGE);
+            } else {
+                $im->flattenImages();
+            }
+
             if (defined('\Imagick::ALPHACHANNEL_OFF')) {
                 // ImageMagick >= 7.0, Imagick >= 3.4.3RC1
                 // @see https://pecl.php.net/package/imagick/3.4.3RC1
                 $im->setImageAlphaChannel(\Imagick::ALPHACHANNEL_OFF);
             } else {
                 $im->setImageAlphaChannel(\Imagick::ALPHACHANNEL_RESET);
             }
-            $im->thumbnailImage($width, $height, true);
 
-            if (method_exists($im, 'mergeImageLayers')) {
-                // Replace flattenImages in imagick 3.3.0
-                // @see https://pecl.php.net/package/imagick/3.3.0RC2
-                $im->mergeImageLayers(\Imagick::LAYERMETHOD_MERGE);
-            } else {
-                $im->flattenImages();
-            }
+            $im->thumbnailImage($width, $height, true);
+            $resource = $this->resourceManager->importResourceFromContent($im->getImageBlob(), $filenameWithoutExtension . '.png');
```

Comment:
Why do you load the resource now twice and destroy twice?

Reply:
Sorry I will fix that soon :)